src/main/java/com/unascribed/fabrication/mixin/g_weird_tweaks/dimensional_tools/MixinItemStack.java

```diff
@@ -0,0 +1,41 @@
+package com.unascribed.fabrication.mixin.g_weird_tweaks.dimensional_tools;
+
+import com.unascribed.fabrication.support.DimensionalTools;
+import com.unascribed.fabrication.support.EligibleIf;
+import com.unascribed.fabrication.support.Env;
+import com.unascribed.fabrication.support.MixinConfigPlugin;
+import net.minecraft.client.item.TooltipContext;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableText;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.List;
+
+@Mixin(ItemStack.class)
+@EligibleIf(configAvailable="*.dimensional_tools")
+public abstract class MixinItemStack {
+
+	@Inject(at=@At("RETURN"), method="getTooltip(Lnet/minecraft/entity/player/PlayerEntity;Lnet/minecraft/client/item/TooltipContext;)Ljava/util/List;")
+	public void getTooltip(PlayerEntity player, TooltipContext context, CallbackInfoReturnable<List<Text>> cir) {
```

Comment:
This is the same injection point as Fabric API's ItemTooltipCallback. It is *not* server-side.

Reply:
ah, in that case the bee thing needs to be changed